### PR TITLE
Split provisioning role

### PR DESCRIPTION
**Pull request by xprazak2**
Split provisioning role into smaller parts
that could be reused downstream.

### CODE REVIEW

File: roles/foreman_provisioning_installer/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+---
+- name: 'install foreman-libvirt'
+  yum: name=foreman-libvirt state=present
```
Comment (ekohl) [inline roles/foreman_provisioning_installer/tasks/main.yml]: It should be sufficient to add ```--enable-foreman-compute-libvirt``` as an installer option

File: roles/ruby_libvirt_gem/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+---
+- name: 'Install libvirt gem'
+  yum: name=tfm-rubygem-ruby-libvirt state=present
```
Comment (ekohl) [inline roles/ruby_libvirt_gem/tasks/main.yml]: I realise this was present before, but is this really needed? ```foreman-libvirt``` depends on this so it should be redundant.

File: roles/foreman_provisioning_installer/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+- name: 'enable provisioning with foreman-installer'
```
Comment (ekohl) [inline roles/foreman_provisioning_installer/tasks/main.yml]: Any chance we can re-use the ```foreman_installer``` role here?

File: roles/foreman_provisioning_infrastructure/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,120 @@
+---
+- name: 'restart foreman-proxy'
```
Comment (ekohl) [inline roles/foreman_provisioning_infrastructure/tasks/main.yml]: Starting with a restart proxy and foreman doesn't make sense to me. Since foreman-proxy doesn't need the libvirt gems I think this can safely be removed.

File: roles/foreman_provisioning_infrastructure/tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+  service: name=foreman-proxy state=restarted
+
+- name: 'restart foreman'
```
Comment (ekohl) [inline roles/foreman_provisioning_infrastructure/tasks/main.yml]: If the installer option is used this can be removed, otherwise it can be moved to where the package is installed.

**Comment by xprazak2** [comment]
Changes:
* using `--enable-foreman-compute-libvirt` now
* reusing `foreman-installer` role
* I kept the ruby-libvirt gem since it is required by libvirt DNS and DHCP providers, but removed the restart tasks since installer is executed and it should do it for us, right?

File: .gitignore [hunk 6]
```diff
@@ -14,7 +14,7 @@ Gemfile.lock
 .ruby-gemset
 *.retry
 *.pyc
-user_playbooks
+user_playbooks/*
```
Comment (ekohl) [inline .gitignore]: Why is this change needed?

File: roles/foreman_installer/defaults/main.yml [hunk 7]
```diff
@@ -4,6 +4,7 @@ foreman_installer_admin_password: changeme
 foreman_installer_skip_installer: False
 foreman_installer_verbose: True
 foreman_installer_scenario: foreman
+foreman_installer_scenario_flag: '{{ "--scenario" if foreman_installer_scenario != "" else "" }}'
```
Comment (ekohl) [inline roles/foreman_installer/defaults/main.yml]: This feels like it should either be in ```install.yml``` or also used in ```upgrade.yml```.

File: roles/foreman_installer/tasks/upgrade.yml [hunk 8]
```diff
@@ -18,6 +18,5 @@
     --upgrade
     --scenario "{{ foreman_installer_scenario }}"
     --certs-update-all
-    {{ foreman_installer_options }}
-    {{ foreman_installer_options_internal_use_only | join(" ") }}
+    {{ foreman_installer_options }} {{ foreman_installer_options_internal_use_only | join(" ") }}
```
Comment (ekohl) [inline roles/foreman_installer/tasks/upgrade.yml]: nit: please revert this so it's consistent with ```install.yaml```

**Comment by xprazak2** [comment]
I think I figured it out. It is supposed to be `rubygem-ruby-libvirt` not `tfm-rubygem-ruby-libvirt` because proxy does not use gems from sw collection.

So I install it, restart proxy and then continue with running installer.

**Comment by ekohl** [comment]
@xprazak2 that makes sense

**Comment by xprazak2** [comment]
@ekohl, I moved the installation of `rubygem-ruby-libvirt` into a separate role, `foreman_installer_scenario_flag` is used in both `install.yml` and `update.yml`. Anything else I should change?

File: roles/foreman_installer/defaults/main.yml (hunk 7 above)
Comment (ekohl) [reply roles/foreman_installer/defaults/main.yml]: (done)

File: roles/restart_proxy/tasks/main.yml [hunk 9]
```diff
@@ -0,0 +1,3 @@
+---
```
Comment (ekohl) [inline roles/restart_proxy/tasks/main.yml]: I wonder if this should be its own role. or if this should be in the ```ruby_libvirt_gem``` role as a trigger so it only restarts when it's installed.

**Comment by xprazak2** [comment]
I moved the proxy restart into a handler.

**Comment by xprazak2** [comment]
I found out katello_provisioning does not work properly, changes will be needed.

**Comment by xprazak2** [comment]
It should work as expected now

**Comment by ares** [comment]
what's the status here? @ekohl approved the PR (there was change afterwards though)

**Comment by ares** [comment]
is there something missing in this PR blocking the merge?

**Comment by xprazak2** [comment]
Rebased and fixed merge conflicts.

**Comment by ehelms** [comment]
Odd, sorry @xprazak2 this is still complaining of needing a rebase

**Comment by xprazak2** [comment]
Rebased.
